src/p2/utils.rs

```diff
@@ -1561,3 +1588,67 @@ where
         line_number,
     })
 }
+
+/// return true if the component with the given name is a markdown component
+/// otherwise returns false
+pub fn is_markdown_component(doc: &ftd::p2::TDoc, name: &str, line_number: usize) -> bool {
+    let mut name = name.to_string();
+    // check if the component is derived from ftd#text
+    while !name.eq("ftd.kernel") {
+        match get_thing_ignore_fail(name.as_str(), line_number, doc) {
+            Some(ftd::p2::Thing::Component(component)) => {
+                if name.eq("ftd#text") {
+                    return true;
+                }
+                name = component.root;
+            }
+            Some(_) => return false,
+            None => return false,
+        }
+    }
+    false
+}
+
+/// fetches the thing from the bag and returns it.
+/// in case of an error while fetching the thing, it returns None
+/// to avoid scenarios when section/subsection don't have a thing in the bag
+/// for eg. in case of ftd#ui
+pub fn get_thing_ignore_fail(
+    name: &str,
+    line_number: usize,
+    doc: &ftd::p2::TDoc,
+) -> Option<ftd::p2::Thing> {
+    match doc.get_thing(line_number, name) {
+        Ok(thing) => Some(thing),
+        Err(_) => None,
+    }
+}
```

Comment:
```suggestion
doc.get_thing(line_number,  name).ok()
```

You don't need `get_thing_ignore_fail` function at all. Use `.ok()` instead

Reply:
@Heulitig FYI